x-pack/metricbeat/module/openai/usage/_meta/fields.yml

```diff
@@ -0,0 +1,233 @@
+        - name: user_id
+          type: keyword
+          description: User identifier
+        - name: organization_id
+          type: keyword
+          description: Organization identifier
+        - name: api_key_id
+          type: keyword
+          description: API key identifier
+        - name: api_key_name
+          type: keyword
+          description: API key name
+        - name: api_key_redacted
+          type: keyword
+          description: Redacted API key
+        - name: api_key_type
+          type: keyword
+          description: Type of API key
+        - name: organization_name
+          type: keyword
+          description: Organization name
+        - name: project_id
+          type: keyword
+          description: Project identifier
+        - name: project_name
+          type: keyword
+          description: Project name
+
+    - name: tts
+      type: group
+      description: >
+        Text-to-Speech API usage metrics
+      fields:
+        - name: timestamp
+          type: date
+          description: Timestamp of request
+        - name: model_id
+          type: keyword
+          description: Model identifier
+        - name: num_characters
+          type: long
+          description: Number of characters processed
+        - name: num_requests
+          type: long
+          description: Number of requests
+        - name: user_id
+          type: keyword
+          description: User identifier
+        - name: organization_id
+          type: keyword
+          description: Organization identifier
+        - name: api_key_id
+          type: keyword
+          description: API key identifier
+        - name: api_key_name
+          type: keyword
+          description: API key name
+        - name: api_key_redacted
+          type: keyword
+          description: Redacted API key
+        - name: api_key_type
+          type: keyword
+          description: Type of API key
+        - name: organization_name
+          type: keyword
+          description: Organization name
+        - name: project_id
+          type: keyword
+          description: Project identifier
+        - name: project_name
+          type: keyword
+          description: Project name
+
+    - name: ft_data
+      type: group
+      description: >
+        Fine-tuning data metrics
+      fields:
+        - name: original
+          type: object
+          object_type: keyword
+          description: Raw fine-tuning data
+
+    - name: assistant_code_interpreter
+      type: group
+      description: >
+        Assistant Code Interpreter usage metrics
+      fields:
+        - name: original
+          type: object
+          object_type: keyword
+          description: Raw assistant code interpreter data
+
+    - name: retrieval_storage
+      type: group
+      description: >
+        Retrieval storage usage metrics
+      fields:
+        - name: original
+          type: object
+          object_type: keyword
```

Comment:
Are all the sub metrics under the object type here keyword ?

Reply:
We never received the data for this. Also didn't find anything anywhere. So that's why it's kept like that. It's also true for ft_data and assistant_code_interpreter